pkg/engine/engine_test.go

```diff
@@ -2694,6 +2699,16 @@ func TestGetAddonFuncMap(t *testing.T) {
 							},
 							OSType: api.Linux,
 						},
+						{
```

Comment:
Added this 2nd agent pool to prove that the `GetAgentPoolZones` go template func will always return a yaml array string representation of the *1st* pool that has a zones configuration. I think the way that csi disk drivers is configured already suggests such mixed-zoned cluster configurations are going to encounter edge case failures, but choosing the first pool is what we've done historically, so this follows that pattern.